From 2be6502951d29aa4f3c217928b8678e9a461cb07 Mon Sep 17 00:00:00 2001
From: Valentine Barshak <valentine.barshak@cogentembedded.com>
Date: Tue, 25 Jun 2019 22:40:40 +0300
Subject: [PATCH 13/22] iommu: ipmmu-vmsa: Add r8a779{7|8}0 whitelist

This adds empty r8a779{7|8}0 whitelist disabling
IPMMU on R-Car V3M and V3H SoCs.

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 drivers/iommu/ipmmu-vmsa.c | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/drivers/iommu/ipmmu-vmsa.c b/drivers/iommu/ipmmu-vmsa.c
index 731a9fc..4fc3208 100644
--- a/drivers/iommu/ipmmu-vmsa.c
+++ b/drivers/iommu/ipmmu-vmsa.c
@@ -230,6 +230,16 @@ static struct ipmmu_whitelist *r8a77965_whitelist[] = {
 	NULL, /* Terminator */
 };
 
+/* R-Car V3M (R8A77970) */
+static struct ipmmu_whitelist *r8a77970_whitelist[] = {
+	NULL, /* Terminator */
+};
+
+/* R-Car V3H (R8A77980) */
+static struct ipmmu_whitelist *r8a77980_whitelist[] = {
+	NULL, /* Terminator */
+};
+
 /* R-Car E3 (R8A77990) */
 static struct ipmmu_whitelist r8a77990_ipmmu_vi0 = {
 	.ipmmu_name	= "febd0000.mmu",
@@ -1050,6 +1060,16 @@ static const struct soc_device_attribute r8a77965[]  = {
 	{ /* sentinel */ }
 };
 
+static const struct soc_device_attribute r8a77970[]  = {
+	{ .soc_id = "r8a77970" },
+	{ /* sentinel */ }
+};
+
+static const struct soc_device_attribute r8a77980[]  = {
+	{ .soc_id = "r8a77980" },
+	{ /* sentinel */ }
+};
+
 static const struct soc_device_attribute r8a77990[]  = {
 	{ .soc_id = "r8a77990" },
 	{ /* sentinel */ }
@@ -1437,6 +1457,10 @@ static int ipmmu_whitelist_init(struct ipmmu_vmsa_device *mmu)
 		mmu->whitelist = r8a77965_whitelist;
 	else if (soc_device_match(r8a77990))
 		mmu->whitelist = r8a77990_whitelist;
+	else if (soc_device_match(r8a77970))
+		mmu->whitelist = r8a77970_whitelist;
+	else if (soc_device_match(r8a77980))
+		mmu->whitelist = r8a77980_whitelist;
 	else
 		mmu->whitelist = NULL;
 
-- 
2.7.4

